### DISCUSSION TITLE
Encountered 1 file that should have been a pointer, but wasn't:

### DISCUSSION THREAD

**Discussion by richban** [git-lfs/git-lfs#5134] [Q&A]
I have migrated an existing repository with:

```shell
git lfs migrate import --everything --above=10Mb
```

when checked out a branch a I got a warning message:

```
Encountered 1 file that should have been a pointer, but wasn't:
```

The file is saved in it's original format (less than 10mb) - it's not a pointer, however it's listed in the `.gitattributes`. 

1. What's the reason it's included in the `.gitattributes` ? Could it be that this file has been more then 10mb previously in the history and `git lfs` included it? 
2. Should I `git track file` or delete it from `gitattributes`?

**Comment by chrisd8088** [comment]
I suspect you are correct that at some point in the history the file's size was greater than 10 MB.  You could examine the history of `.gitattributes` to find the commit where the relevant line was added (e.g., with `git blame .gitattributes`), and then look at that commit to see what the file's size was at that time.

```
$ git init above
$ cd above
$ echo 1234567890 > foo.bin
$ git add foo.bin
$ git commit -m ten
$ echo 12345678901234567890 > foo.bin 
$ git add foo.bin
$ git commit -m twenty
$ git show HEAD^:foo.bin > foo.bin
$ git add foo.bin
$ git commit -m 'ten again'

$ git lfs migrate import --everything --above=15b
$ cat .gitattributes
/foo.bin filter=lfs diff=lfs merge=lfs -text
$ git lfs fsck
pointer: unexpectedGitObject: "foo.bin" (treeish 1f01798220ed10885e5b6622f2f6384607ea79bd) should have been a pointer but was not

$ git blame .gitattributes
ce1543be (Chris Darroch 2022-10-04 20:23:37 -0700 1) /foo.bin filter=lfs diff=lfs merge=lfs -text

$ git show --oneline ce1543be 
ce1543b twenty
diff --git a/.gitattributes b/.gitattributes
new file mode 100644
index 0000000..daca9d7
--- /dev/null
+++ b/.gitattributes
@@ -0,0 +1 @@
+/foo.bin filter=lfs diff=lfs merge=lfs -text
diff --git a/foo.bin b/foo.bin
index a32a434..677de18 100644
--- a/foo.bin
+++ b/foo.bin
@@ -1 +1,3 @@
-1234567890
+version https://git-lfs.github.com/spec/v1
+oid sha256:8a6a84214f1518875819761ffb53d33ec19e66a089213b906720f6825715e8b3
+size 21

$ git show --oneline HEAD
1f01798 (HEAD -> main) ten again
diff --git a/foo.bin b/foo.bin
index 677de18..a32a434 100644
--- a/foo.bin
+++ b/foo.bin
@@ -1,3 +1 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:8a6a84214f1518875819761ffb53d33ec19e66a089213b906720f6825715e8b3
-size 21
+1234567890
```

As an immediate fix here, one could remove the line from `.gitattributes` or replace it with a line like `/foo.bin !text !filter !merge !diff` in the commits where the size of the file drops below your 10 MB limit.  Obviously you'd want to rebase to squash those changes into the appropriate places in history.  In the very simple case of the example above, this could just be:
```
$ git rm .gitattributes
$ git commit --amend -C HEAD
```

More generally, though, I think this likely represents a bug in how `git lfs migrate import --above` is supposed to work.  I notice at least one issue comment in https://github.com/git-lfs/git-lfs/issues/5021#issuecomment-1141283495 already discusses this concern, so I'm going to copy the example above into another comment in that issue for reference.
